docs/standard/serialization/binaryformatter-migration-guide/migrate-to-system-text-json.md

```diff
@@ -0,0 +1,51 @@
+---
+title: "BinaryFormatter migration guide: Migrate to System.Text.Json (JSON)"
+description: "Migrate from BinaryFormatter to System.Text.Json for JSON serialization."
+ms.date: 5/31/2024
+no-loc: [BinaryFormatter, Serialization, System.Text.Json]
+dev_langs:
+  - CSharp
+helpviewer_keywords:
+  - "BinaryFormatter"
+  - "System.Text.Json"
+  - "serializing objects"
+  - "serialization"
+  - "objects, serializing"
+---
+
+# Migrate to System.Text.Json (JSON)
+
+[System.Text.Json](../system-text-json/overview.md) defaults to emphasizing literal, deterministic behavior and avoids any guessing or interpretation on the caller's behalf. The library is intentionally designed this way for security and performance. From the migration perspective, it's crucial to know the following facts:
+
+- By default, **fields are not serialized or deserialized**, but they can be [annotated for serialization](../system-text-json/fields.md), or `JsonSerializerOptions.IncludeFields` can be cautiously set to `true` to include all fields.
+
+  ```csharp
+  JsonSerializerOptions options = new()
+  {
+      IncludeFields = true
+  };
+  ```
+
+- By default, System.Text.Json **ignores private fields and properties**. You can enable use of a non-public accessor on a property by using the `[JsonInclude]` attribute. Including private fields requires some [non-trivial extra work](../system-text-json/custom-contracts.md#example-serialize-private-fields).
+
+- System.Text.Json **[cannot deserialize read-only fields](/dotnet/api/system.text.json.jsonserializeroptions.ignorereadonlyfields?view#remarks)** or properties, but the `[JsonConstructor]` attribute can be used to indicate that the specified constructor should be used to create instances of the type on deserialization. The constructor can set the read-only fields and properties.
+
+- It [supports serialization and deserialization of most built-in collections](../system-text-json/supported-collection-types.md). The exceptions:
+  - multi-dimensional arrays,
+  - `BitArray`,
+  - `LinkedList<T>`,
```

Comment:
I just checked and `LinkedList<T>` seems to be working fine.

Reply:
Do `System.Collections.Immutable` work? If not, we should probably list those.